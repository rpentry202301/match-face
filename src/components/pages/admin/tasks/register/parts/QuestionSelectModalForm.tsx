"use client";
import { useState, ReactNode, ChangeEvent, FormEvent, useMemo } from "react";
import { createPortal } from "react-dom";
import WhiteButton from "@/components/ui/button/WhiteButton";
import WhiteButtonCheckBox from "./WhiteButtonCheckBox";
import OrangeButton from "@/components/ui/button/OrangeButton";
import CheckBox from "@/components/ui/checkbox/CheckBox";
import QuestionList from "./QuestionList";
import {
  Questions,
  useSelectedQuestion,
} from "@/hooks/store/context/SelectedQuestionContext";
import type { FetchQuestionModalData } from "@/types/admin/tasks/register/types";
import SelectProjects from "./SelectProjects";

// 実際にレンダリングされるモーダルは以下に記述
const QuestionSelectModalForm = ({
  fetchData,
}: {
  fetchData: FetchQuestionModalData;
}) => {
  const [search, setSearch] = useState("");
  const [isOpened, setIsOpened] = useState(false);
  const [questions, setQuestions] = useState(fetchData.questions);

  // 状態初期化用にオブジェクトを作成
  const initDepartments = fetchData.departments.map((dep) => {
    return {
      id: dep.id,
      label: dep.name,
      checked: false,
    };
  });

  const initSkills = fetchData.skills.map((skill) => {
    return {
      id: skill.id,
      label: skill.name,
      checked: false,
    };
  });

  const [formData, setFormData] = useState({
    search: "",
    department: initDepartments,
    skill: initSkills,
  });

  const [selectedQuestion, selectedQuestionDispatch] = useSelectedQuestion();
  const [checkedValues, setCheckedValue] =
    useState<Questions>(selectedQuestion);

  // 選択中のprojectsデータID
  const [activePj, setActivePj] = useState({
    id: 0,
    name: "",
  });
  /**
   * activePj切替用ハンドラ
   */
  const handleChangePj = (id: number, name: string) => {
    if (id === activePj.id) {
      return;
    } else {
      setActivePj({ id: id, name: name });
      setCheckedValue({ projectId: id, list: [] });
      selectedQuestionDispatch({
        type: "select",
        payload: { projectId: id, list: [] },
      });
    }
  };
  // console.log("activePj", activePj);
  // console.log("checkedValues", checkedValues);
  // console.log("selectedQuestion", selectedQuestion);

  // 現在の案件に一致する質問データ
  const activePjQuestions = useMemo(() => {
    return questions.filter((question) => question.projectId === activePj.id);
  }, [questions, activePj]);

  const open = () => {
    setCheckedValue(selectedQuestion);
    setIsOpened(true);
  };
  const close = () => setIsOpened(false);

  const handleChangeCheckBox = (
    e: ChangeEvent<HTMLInputElement>,
    name: "department" | "skill"
  ) => {
    const newData = formData[name];
    newData.map((data) => {
      if (data.label === e.target.value) {
        data.checked = !data.checked;
      }
      return newData;
    });
    setFormData({ ...formData, [name]: newData });
  };

  const handleChangeQuestionList = (
    e: ChangeEvent<HTMLInputElement>,
    id: number
  ) => {
    if (checkedValues.list.map((Q) => Q.name).includes(e.target.value)) {
      setCheckedValue({
        ...checkedValues,
        list: checkedValues.list.filter(
          (question) => question.name !== e.target.value
        ),
      });
    } else {
      setCheckedValue({
        ...checkedValues,
        list: [...checkedValues.list, { id: id, name: e.target.value }],
      });
    }
  };

  const handleClose = () => {
    selectedQuestionDispatch({ type: "select", payload: checkedValues });
    close();
  };

  // Todo: APIができたら実装
  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    const searchKeyword = search;
    const checkedDepId = formData.department
      .filter((data) => data.checked)
      .map((dep) => dep.id);
    const checkedSkillId = formData.skill
      .filter((data) => data.checked)
      .map((skill) => skill.id);

    const searchQuerys = [
      searchKeyword ? `searchKeyword=${searchKeyword}` : "",
      checkedDepId.length ? `departmentId=${checkedDepId}` : "",
      checkedSkillId.length ? `skillId=${checkedSkillId}` : "",
    ];

    const query = searchQuerys.filter((query) => query !== "").join("&");

    fetch(`/api/admin/tasks/register/questions?${query}`, {
      cache: "no-cache",
      method: "GET",
      headers: {
        "Content-Type": "application/json",
      },
    })
      .then((res) => {
        console.log(res.status);
        return res.json();
      })
      .then((data) => {
        setQuestions(data.questionList);
      })
      .catch((err) => console.log(err));
  };

  return (
    <Modal buttonText="追加" isOpened={isOpened} open={open} close={close}>
      <div className="flex flex-col items-center gap-8 w-full mx-auto">
        <div className="mx-auto">
          <div className="flex flex-col items-start gap-3">
            <div className="text-base">
              <h2>▶️質問を選択する</h2>
            </div>
            <form
              onSubmit={handleSubmit}
              className="flex flex-col item-center justify-center px-12 py-5 border-2 w-full"
            >
              <div className="flex flex-col items-center gap-5">
                <SelectProjects
                  projects={fetchData.projects}
                  activePj={activePj}
                  onClick={(id: number, name: string) =>
                    handleChangePj(id, name)
                  }
                />
                <div className="flex flex-col items-center gap-5">
                  <div className="flex items-start justify-center gap-2 w-fit">
                    <input
                      id="search"
                      type="text"
                      value={search}
                      onChange={(e) => setSearch(e.target.value)}
                      className="border-light-gray border-2 text-xs p-2 lg:w-96 sm:w-80 w-72"
                      data-testid="search-box"
                    />
                    <WhiteButton label="検索" className="text-xs ml-4 w-16" />
                  </div>
                  <div className="flex items-start gap-4 flex-wrap">
                    {fetchData.departments.map((element) => (
                      <WhiteButtonCheckBox
                        key={`teck_${element.id}`}
                        id={`teck_${element.id}`}
                        label={element.name}
                        value={element.name}
                        checked={
                          formData.department.filter(
                            (data) => data.label === element.name
                          )[0].checked
                        }
                        name="department"
                        className="text-xs w-16"
                        onChange={(e) => handleChangeCheckBox(e, "department")}
                      />
                    ))}
                  </div>
                  <div className="flex mx-auto gap-4">
                    <span className="text-xs">使用技術：</span>
                    <div className="grid xl:grid-cols-7 lg:grid-cols-6 md:grid-cols-4 grid-cols-3 gap-3">
                      {fetchData.skills.map((skill) => (
                        <div key={`skill_${skill.id}`} className="flex">
                          <CheckBox
                            id={`skill_${skill.id}`}
                            value={skill.name}
                            checked={
                              formData.skill.filter(
                                (data) => data.label === skill.name
                              )[0].checked
                            }
                            name="skill"
                            onChange={(e) => handleChangeCheckBox(e, "skill")}
                          />
                          <label
                            htmlFor={`skill_${skill.id}`}
                            className="text-xs ml-1 hover:cursor-pointer"
                          >
                            {skill.name}
                          </label>
                        </div>
                      ))}
                    </div>
                  </div>
                </div>
                <OrangeButton
                  type="submit"
                  label="絞り込み"
                  className="text-xs"
                />
              </div>
            </form>
          </div>
        </div>

        {activePj.id === 0 ? (
          <div>案件名を選択して下さい</div>
        ) : activePjQuestions.length === 0 ? (
          <div>表示できる質問がありません</div>
        ) : (
          <>
            <QuestionList
              questions={activePjQuestions}
              checkedValues={checkedValues}
              onChange={handleChangeQuestionList}
            />

            <OrangeButton
              label="選択完了"
              className="text-xs"
              onClick={handleClose}
            />
          </>
        )}
        {/* <QuestionList
          questions={questions}
          checkedValues={checkedValues}
          onChange={handleChangeQuestionList}
        />

        <OrangeButton
          label="選択完了"
          className="text-xs"
          onClick={handleClose}
        /> */}
      </div>
    </Modal>
  );
};

const Modal = ({
  children,
  buttonText,
  canCloseByClickingBackground = true,
  isOpened,
  open,
  close,
}: {
  children: ReactNode;
  buttonText: string;
  canCloseByClickingBackground?: boolean;
  isOpened: boolean;
  open: () => void;
  close: () => void;
}) => {
  if (!isOpened) {
    return (
      <WhiteButton label={buttonText} onClick={open} className="text-xs" />
    );
  }

  // レンダリングするDOMをbodyに固定するためPortalを使用
  const elmModal = (
    <div className="fixed top-0 left-0 z-20 flex items-center justify-center w-full h-full">
      <div className="relative z-10 w-3/4 max-w-5xl py-7 px-10 bg-white">
        {children}
      </div>
      {canCloseByClickingBackground ? (
        <div
          className="absolute top-0 left-0 w-full h-full bg-black opacity-30"
          onClick={close}
        />
      ) : (
        <div className="absolute top-0 left-0 w-full h-full bg-black opacity-30" />
      )}
    </div>
  );
  return createPortal(elmModal, document.body);
};

export default QuestionSelectModalForm;
