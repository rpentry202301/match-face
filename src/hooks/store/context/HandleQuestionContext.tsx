"use client";

import React, { useContext } from "react";

type Refine = {
  word: string[];
  department: number[];
};

const SelectContext = React.createContext<[Refine, React.Dispatch<React.SetStateAction<Refine>>] | undefined>(undefined);

export function SelectProvider({ children }: { children: React.ReactNode }) {
  const [select, setSelect] = React.useState<Refine>({
    word: [],
    department: [],
  });

  return (
    <SelectContext.Provider value={[select, setSelect]}>
      {children}
    </SelectContext.Provider>
  );
}

export function useRefine() {
  const context = useContext(SelectContext);
  if (context === undefined) {
    throw new Error("useRefine must be used within a SelectProvider");
  }
  return context;
}
