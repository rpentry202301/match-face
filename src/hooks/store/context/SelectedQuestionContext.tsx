"use client";
import {
  Dispatch,
  ReactNode,
  Reducer,
  createContext,
  useContext,
  useReducer,
} from "react";
import { QusetionsReducer } from "../reducer/selecterReducer";
import { QuestionsReducerAction } from "@/types/admin/tasks/register/types";

export type Questions = {
  projectId: number;
  list: {
    id: number;
    name: string;
  }[];
};

const SelectedQuestionContext = createContext<
  [Questions, Dispatch<QuestionsReducerAction>] | undefined
>(undefined);

export const SelectedQuestionProvider = ({
  children,
}: {
  children: ReactNode;
}) => {
  const [state, dispatch] = useReducer<
    Reducer<Questions, QuestionsReducerAction>
  >(QusetionsReducer, { projectId: 0, list: [] });
  return (
    <SelectedQuestionContext.Provider value={[state, dispatch]}>
      {children}
    </SelectedQuestionContext.Provider>
  );
};

export const useSelectedQuestion = () => {
  const context = useContext(SelectedQuestionContext);
  if (context === undefined)
    throw new Error("Function 'useSelectedQuestion' must be used in Provider.");
  return context;
};
