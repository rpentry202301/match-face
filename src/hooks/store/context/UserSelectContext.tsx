"use client";
import {
  Dispatch,
  ReactNode,
  Reducer,
  createContext,
  useContext,
  useReducer,
} from "react";
import { UsersReducer } from "../reducer/selecterReducer";
import { UsersReducerAction } from "@/types/admin/tasks/register/types";

export type Users = {
  id: string;
  name: string;
};

const UserSelectContext = createContext<
  [Users[], Dispatch<UsersReducerAction>] | undefined
>(undefined);

export const UserSelectProvider = ({ children }: { children: ReactNode }) => {
  const [state, dispatch] = useReducer<Reducer<Users[], UsersReducerAction>>(
    // selecterReducer,
    UsersReducer,
    []
  );
  return (
    <UserSelectContext.Provider value={[state, dispatch]}>
      {children}
    </UserSelectContext.Provider>
  );
};

export const useUserSelect = () => {
  const context = useContext(UserSelectContext);
  if (context === undefined)
    throw new Error("Function 'useUserSelect' must be used in Provider.");
  return context;
};
