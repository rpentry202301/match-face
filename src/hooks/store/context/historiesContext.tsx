'use client'
import { createContext, Dispatch, useContext, useReducer } from "react";
import { SelectHistoryAction } from "@/types/admin/histories/admin_histories";
import { SelectHistoryReducer } from "../reducer/historiesReducer";

const SelectHistoryContext = createContext<{month:string,department:number[],skills:number[]}| undefined>(undefined);
const SelectHistoryDispatchContext = createContext<Dispatch<SelectHistoryAction>|undefined>(undefined)
const initialSelect = {month:"",department:[],skills:[]}

export function SelectHistoryProvider ({children}:{children:React.ReactNode}){
    const [state,dispatch] = useReducer(SelectHistoryReducer,initialSelect)
    return (
        <SelectHistoryContext.Provider value={state}>
            <SelectHistoryDispatchContext.Provider value={dispatch} >
            {children}
            </SelectHistoryDispatchContext.Provider>
        </SelectHistoryContext.Provider>
    )
}

export const useSelectHistory = () => {
    const context = useContext(SelectHistoryContext)
    if(context === undefined) {
        throw new Error("Function 'useSelectHistory' must be used in Provider.")
    }
    return context
}

export const useSelectHistoryDispatch = () => {
    const dispatch = useContext(SelectHistoryDispatchContext)
    if(dispatch === undefined) {
        throw new Error("Function 'useSelectHistoryDispatch' must be used in Provider.")
    }
    return dispatch
}
