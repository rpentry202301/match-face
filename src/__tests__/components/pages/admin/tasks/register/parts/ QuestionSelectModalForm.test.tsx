import QuestionSelectModalForm from "@/components/pages/admin/tasks/register/parts/QuestionSelectModalForm";
import { SelectedQuestionProvider } from "@/hooks/store/context/SelectedQuestionContext";
import { screen, render, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

export const fetchData = {
  departments: [
    {
      id: 1,
      name: 'Java',
      createdUser: '初期設定',
      createdAt: '2023-09-07T09:29:00.130462',
      updateUser: '初期設定',
      updateAt: '2023-09-07T09:29:00.130462'
    },
    {
      id: 2,
      name: 'PHP',
      createdUser: '初期設定',
      createdAt: '2023-09-07T09:29:00.130462',
      updateUser: '初期設定',
      updateAt: '2023-09-07T09:29:00.130462'
    },
    {
      id: 3,
      name: 'FR',
      createdUser: '初期設定',
      createdAt: '2023-09-07T09:29:00.130462',
      updateUser: '初期設定',
      updateAt: '2023-09-07T09:29:00.130462'
    },
    {
      id: 4,
      name: 'CL',
      createdUser: '初期設定',
      createdAt: '2023-09-07T09:29:00.130462',
      updateUser: '初期設定',
      updateAt: '2023-09-07T09:29:00.130462'
    },
    {
      id: 5,
      name: 'ML',
      createdUser: '初期設定',
      createdAt: '2023-09-07T09:29:00.130462',
      updateUser: '初期設定',
      updateAt: '2023-09-07T09:29:00.130462'
    },
    {
      id: 6,
      name: 'QA',
      createdUser: '初期設定',
      createdAt: '2023-09-07T09:29:00.130462',
      updateUser: '初期設定',
      updateAt: '2023-09-07T09:29:00.130462'
    }
  ],
  skills: [
    {
      id: 1,
      name: 'JavaScript',
      createdUser: 'テスト用',
      createdAt: '2023-09-07T09:30:18.99969',
      updateUser: 'テスト用',
      updateAt: '2023-09-07T09:30:18.99969'
    },
    {
      id: 2,
      name: 'TypeScript',
      createdUser: 'テスト用',
      createdAt: '2023-09-07T09:30:18.99969',
      updateUser: 'テスト用',
      updateAt: '2023-09-07T09:30:18.99969'
    },
    {
      id: 3,
      name: 'React',
      createdUser: 'テスト用',
      createdAt: '2023-09-07T09:30:18.99969',
      updateUser: 'テスト用',
      updateAt: '2023-09-07T09:30:18.99969'
    },
    {
      id: 4,
      name: 'Next.js',
      createdUser: 'テスト用',
      createdAt: '2023-09-07T09:30:18.99969',
      updateUser: 'テスト用',
      updateAt: '2023-09-07T09:30:18.99969'
    },
  ],
  questions: [
    {
      id: 1,
      projectId: 1,
      projectName: 'バックエンド案件',
      context: '開発経験について教えてください。',
      createdUser: '',
      createdAt: '-999999999-01-01T00:00:00',
      updateUser: 'テスト花子',
      updateAt: '2023-09-07T09:29:24.412579'
    },
    {
      id: 2,
      projectId: 1,
      projectName: 'バックエンド案件',
      context: '1番得意な言語は何ですか。',
      createdUser: '',
      createdAt: '-999999999-01-01T00:00:00',
      updateUser: 'テスト花子',
      updateAt: '2023-09-07T09:29:24.412579'
    }
  ]
}

// モーダル展開用関数
const open = () => {
  render(
    <SelectedQuestionProvider>
      <QuestionSelectModalForm fetchData={fetchData} />
    </SelectedQuestionProvider>
  );
  fireEvent.click(screen.getByRole("button"));
}

describe("QuestionSelectModal.tsx", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    open();
  })

  describe("スナップショットテスト", () => {
    it("モーダルを開いている状態", () => {
      const { baseElement } = render(
        <SelectedQuestionProvider>
          <QuestionSelectModalForm fetchData={fetchData} />
        </SelectedQuestionProvider>
      );
      expect(baseElement).toMatchSnapshot();
    });
  })

  describe("入力テスト", () => {
    it("検索ボックスの入力テスト", async () => {
      render(
        <SelectedQuestionProvider>
          <QuestionSelectModalForm fetchData={fetchData} />
        </SelectedQuestionProvider>
      );
      const input = screen.getByTestId("search-box") as HTMLInputElement;
      fireEvent.change(input, { target: { value: "案件" } });
      expect(input.value).toBe("案件");
    });
  })

  it("部署のチェックボックステスト", async () => {
    const user = userEvent.setup()
    render(
      <SelectedQuestionProvider>
        <QuestionSelectModalForm fetchData={fetchData} />
      </SelectedQuestionProvider>
    );
    const beforeCheck = screen.getByTestId("check-box:FR") as HTMLInputElement;
    expect(beforeCheck.checked).toBe(false);
    // Javaを選択してテスト
    await user.click(screen.getByTestId("check-box:FR"))
    const afterCheck = screen.getByTestId("check-box:FR") as HTMLInputElement;
    expect(afterCheck.checked).toBe(true);
  })
})
