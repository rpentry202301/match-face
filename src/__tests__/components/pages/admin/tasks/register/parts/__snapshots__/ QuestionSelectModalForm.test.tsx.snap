// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionSelectModal.tsx スナップショットテスト モーダルを開いている状態 1`] = `
<body>
  <div />
  <div
    class="fixed top-0 left-0 z-30 flex items-center justify-center w-full h-full"
  >
    <div
      class="relative z-20 w-3/4 max-w-5xl py-7 px-10 bg-white"
    >
      <div
        class="flex flex-col items-center gap-8 w-full mx-auto"
      >
        <div
          class="mx-auto"
        >
          <div
            class="flex flex-col items-start gap-3"
          >
            <div
              class="text-base"
            >
              <h2>
                ▶️質問を選択する
              </h2>
            </div>
            <form
              class="flex flex-col item-center justify-center px-12 py-5 border-2 w-full"
            >
              <div
                class="flex flex-col items-center gap-5"
              >
                <div
                  class="flex flex-col items-center gap-5"
                >
                  <div
                    class="flex items-start justify-center gap-2 w-fit"
                  >
                    <input
                      class="border-light-gray border-2 text-xs p-2 lg:w-96 sm:w-80 w-72"
                      data-testid="search-box"
                      id="search"
                      type="text"
                      value=""
                    />
                    <button
                      class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md text-xs ml-4 w-16"
                    >
                      検索
                    </button>
                  </div>
                  <div
                    class="flex items-start gap-4 flex-wrap"
                  >
                    <div>
                      <input
                        class="peer hidden"
                        data-testid="check-box:Java"
                        id="teck_1"
                        name="department"
                        type="checkbox"
                        value="Java"
                      />
                      <label
                        class="text-xs inline-flex items-center justify-center py-2 px-4 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                        for="teck_1"
                      >
                        Java
                      </label>
                    </div>
                    <div>
                      <input
                        class="peer hidden"
                        data-testid="check-box:PHP"
                        id="teck_2"
                        name="department"
                        type="checkbox"
                        value="PHP"
                      />
                      <label
                        class="text-xs inline-flex items-center justify-center py-2 px-4 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                        for="teck_2"
                      >
                        PHP
                      </label>
                    </div>
                    <div>
                      <input
                        class="peer hidden"
                        data-testid="check-box:FR"
                        id="teck_3"
                        name="department"
                        type="checkbox"
                        value="FR"
                      />
                      <label
                        class="text-xs inline-flex items-center justify-center py-2 px-4 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                        for="teck_3"
                      >
                        FR
                      </label>
                    </div>
                    <div>
                      <input
                        class="peer hidden"
                        data-testid="check-box:CL"
                        id="teck_4"
                        name="department"
                        type="checkbox"
                        value="CL"
                      />
                      <label
                        class="text-xs inline-flex items-center justify-center py-2 px-4 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                        for="teck_4"
                      >
                        CL
                      </label>
                    </div>
                    <div>
                      <input
                        class="peer hidden"
                        data-testid="check-box:ML"
                        id="teck_5"
                        name="department"
                        type="checkbox"
                        value="ML"
                      />
                      <label
                        class="text-xs inline-flex items-center justify-center py-2 px-4 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                        for="teck_5"
                      >
                        ML
                      </label>
                    </div>
                    <div>
                      <input
                        class="peer hidden"
                        data-testid="check-box:QA"
                        id="teck_6"
                        name="department"
                        type="checkbox"
                        value="QA"
                      />
                      <label
                        class="text-xs inline-flex items-center justify-center py-2 px-4 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                        for="teck_6"
                      >
                        QA
                      </label>
                    </div>
                  </div>
                  <div
                    class="flex mx-auto gap-4"
                  >
                    <span
                      class="text-xs"
                    >
                      使用技術：
                    </span>
                    <div
                      class="grid xl:grid-cols-7 lg:grid-cols-6 md:grid-cols-4 grid-cols-3 gap-3"
                    >
                      <div
                        class="flex"
                      >
                        <input
                          class="w-4 h-4 accent-orange cursor-pointer"
                          id="skill_1"
                          name="skill"
                          type="checkbox"
                          value="JavaScript"
                        />
                        <label
                          class="text-xs ml-1 hover:cursor-pointer"
                          for="skill_1"
                        >
                          JavaScript
                        </label>
                      </div>
                      <div
                        class="flex"
                      >
                        <input
                          class="w-4 h-4 accent-orange cursor-pointer"
                          id="skill_2"
                          name="skill"
                          type="checkbox"
                          value="TypeScript"
                        />
                        <label
                          class="text-xs ml-1 hover:cursor-pointer"
                          for="skill_2"
                        >
                          TypeScript
                        </label>
                      </div>
                      <div
                        class="flex"
                      >
                        <input
                          class="w-4 h-4 accent-orange cursor-pointer"
                          id="skill_3"
                          name="skill"
                          type="checkbox"
                          value="React"
                        />
                        <label
                          class="text-xs ml-1 hover:cursor-pointer"
                          for="skill_3"
                        >
                          React
                        </label>
                      </div>
                      <div
                        class="flex"
                      >
                        <input
                          class="w-4 h-4 accent-orange cursor-pointer"
                          id="skill_4"
                          name="skill"
                          type="checkbox"
                          value="Next.js"
                        />
                        <label
                          class="text-xs ml-1 hover:cursor-pointer"
                          for="skill_4"
                        >
                          Next.js
                        </label>
                      </div>
                    </div>
                  </div>
                </div>
                <div>
                  <button
                    class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white rounded-xl p-2 w-40 h-auto text-xs"
                    type="submit"
                  >
                    絞り込み
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          class="w-full mx-auto"
        >
          <table
            class="border-collapse border-2 border-deep-gray text-sm mx-auto w-full"
          >
            <thead
              class="bg-light-gray"
            >
              <tr>
                <th
                  class="border-2 border-deep-gray px-4 py-2 text-center w-1/12"
                />
                <th
                  class="border-2 border-deep-gray px-4 py-2 text-center w-2/12"
                >
                  最終編集日
                </th>
                <th
                  class="border-2 border-deep-gray px-4 py-2 text-center w-3/12"
                >
                  案件名
                </th>
                <th
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  質問内容
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  <input
                    class="w-4 h-4 accent-orange cursor-pointer"
                    type="checkbox"
                    value="開発経験について教えてください。"
                  />
                </td>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  2023-09-07
                </td>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  バックエンド案件
                </td>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  開発経験について教えてください。
                </td>
              </tr>
              <tr>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  <input
                    class="w-4 h-4 accent-orange cursor-pointer"
                    type="checkbox"
                    value="1番得意な言語は何ですか。"
                  />
                </td>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  2023-09-07
                </td>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  バックエンド案件
                </td>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  1番得意な言語は何ですか。
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div>
          <button
            class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white rounded-xl p-2 w-40 h-auto text-xs"
          >
            選択完了
          </button>
        </div>
      </div>
    </div>
    <div
      class="absolute top-0 left-0 w-full h-full bg-black opacity-30"
    />
  </div>
  <div>
    <button
      class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md text-xs"
    >
      追加
    </button>
  </div>
</body>
`;
